src/main/java/mate/academy/internetshop/dao/hibernate/UserDaoHibernateImpl.java

```diff
@@ -0,0 +1,125 @@
+            user.setSalt(salt);
+            userId = (Long) session.save(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        user.setId(userId);
+        Bucket bucket = new Bucket(user);
+        bucket.setUser(user);
+        bucketDao.create(bucket);
+        return user;
+    }
+
+    @Override
+    public User get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            User user = session.get(User.class, id);
+            return user;
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            session.update(user);
+            transaction.commit();
+        } catch (Exception e) {
+            logger.error("Can't update user " + user, e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return user;
+    }
+
+    @Override
+    public void delete(Long id) {
+        User user = get(id);
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            session.delete(user);
+            transaction.commit();
+        } catch (Exception e) {
+            logger.error("Can't delete user " + user, e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+    }
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("from User where login=:login");
+            query.setParameter("login", login);
+            User user = (User) query.uniqueResult();
+            if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<User> getByToken(String token) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("from User where token=:token");
+            query.setParameter("token", token);
+            List<User> list = query.list();
+            return list.stream().findFirst();
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createCriteria(User.class).list();
+        }
+    }
+
+    @Override
+    public void setRoles(User newUser, Set<Role> rolesFromDb) {
```

Comment:
@Kokhanevich If you are not using this method - please remove. Or use it in a proper way